pymc_marketing/model_builder.py

```diff
@@ -616,7 +616,7 @@ def sample_posterior_predictive(self, X_pred, extend_idata, combined, **kwargs):
         with self.model:  # sample with new input data
             post_pred = pm.sample_posterior_predictive(self.idata, **kwargs)
             if extend_idata:
-                self.idata.extend(post_pred)
+                self.idata.extend(post_pred, join="right")
```

Comment:
Same here. Maybe I missed a spot, so let me know

Reply:
I saw those. What are you referring to? I mentioned `fit` will still override the previous result, which is not something that needs to be fixed in this PR.

The test has the wrong assert because it's checking that the idata matches the first call (not the second). The test is also insufficient, because it doesn't raise (I think you didn't change it when you switched from "left" to "right"

Reply:
I modified the tests in order to check for the second sample stays. Please take a look